mathex/mathex.go

```diff
@@ -0,0 +1,90 @@
+package mathex
+
+import (
+	"math"
+
+	"github.com/EliCDavis/vector"
+	"golang.org/x/exp/constraints"
+)
+
+type Number interface {
+	constraints.Integer | constraints.Float | vector.Number
+}
+
+func NearZero[T Number](v T) bool {
+	const s = 1e-8
+	return Abs(float64(v)) < s
```

Comment:
To prevent more casting and rely less on function inlining

```suggestion
	return math.Abs(float64(v)) < s
```